tests/integration_tests/instances.py

```diff
@@ -223,18 +224,35 @@ def install_ppa(self):
             ).ok
         pin_origin = self.settings.CLOUD_INIT_SOURCE[4:]  # Drop leading ppa:
         pin_origin = re.sub("[^a-z0-9-]", "-", pin_origin)
+        preferences = """\
```

Comment:
F for the win. Otherwise we create bogus apt preferences files which try to match `{pin_origin}`
```suggestion
        preferences = f"""\
```

Reply:
Good catch, thanks! Fixed.